meson.build

```diff
@@ -0,0 +1,90 @@
+project(
+    'com.github.parnold-x.nasc',
+    'vala', 'c', 'cpp',
+    version: '0.5.0',
+    default_options: ['cpp_std=c++11']
+)
+
+gettext_name = 'nasc'
+
+if meson.get_compiler('vala').version() < '0.28.0'
+    error('vala compiler version 0.28.0 or newer is required.')
+endif
+
+pkg_data_dir = join_paths(
+    get_option('prefix'),
+    get_option('datadir'),
+    'nasc'
+)
+
+style_dir = join_paths(
+    get_option('prefix'),
+    get_option('datadir'),
+    'qalculate',
+    'styles'
+)
+
+add_project_arguments(
+    '-DGETTEXT_PACKAGE="@0@"'.format(gettext_name),
+    language: ['c', 'cpp']
+)
+
+conf_data = configuration_data()
+conf_data.set('DATADIR', join_paths(get_option('prefix'), get_option('datadir')))
+conf_data.set('STYLEDIR', style_dir)
+conf_data.set('PKGDATADIR', pkg_data_dir)
+conf_data.set('GETTEXT_PACKAGE', gettext_name)
+conf_data.set('RELEASE_NAME', meson.project_name())
+conf_data.set('VERSION', meson.project_version())
+conf_data.set('VERSION_INFO', 'Release')
+
+config_header = configure_file(
+    input: 'src/config.vala.in',
+    output: 'config.vala',
+    configuration: conf_data
+)
+
+executable(
+    meson.project_name(),
+
+    'src/Calculator.vala',
+    'src/Controller.vala',
+    'src/HelpBox.vala',
+    'src/InputView.vala',
+    'src/ListFooter.vala',
+    'src/MainWindow.vala',
+    'src/Nasc.vala',
+    'src/NascSettings.vala',
+    'src/OpenBox.vala',
+    'src/PasteBinDialog.vala',
+    'src/PeriodicTable.vala',
+    'src/ResultBoxWidget.vala',
+    'src/ResultLine.vala',
+    'src/ResultView.vala',
+    'src/Tutorial.vala',
+
+    config_header,
+
+    'libqalculatenasc/QalculateNasc.cc',
+    include_directories: [include_directories('libqalculatenasc')],
+
+    dependencies: [
+        dependency('gee-0.8'),
+        dependency('glib-2.0'),
+        dependency('granite'),
+        dependency('gtk+-3.0'),
+        dependency('gtksourceview-3.0', version: '>=3.10'),
+        dependency('libsoup-2.4'),
+        meson.get_compiler('c').find_library('libqalculate'),
+        meson.get_compiler('vala').find_library('posix'),
+        meson.get_compiler('vala').find_library('QalculateNasc', dirs: [join_paths(meson.current_source_dir(), 'vapi')])
+    ],
+
+    install: true
+)
+
+meson.add_install_script('meson/post_install.py')
```

Comment:
Do not rely on python in shebang – that will break on NixOS, BSD and other platforms.

Use
```meson
python = import('python')
meson.add_install_script(python.find_installation('python3'), 'meson/post_install.py')
```

Reply:
No. On NixOS, the only thing in `/usr/bin` is `env` (for compatibility), not sure about BSD but see [similar change](https://github.com/hughsie/libxmlb/pull/5) in libxmlb.

Reply:
This doesn't work: `meson.build:106:6: ERROR:  add_install_script args must be strings`

Reply:
I've changed it to use `#!/usr/bin/env python3` instead.

Reply:
Try `.path()` to the the installation.

Reply:
Doesnt' work. The object doesn't have a `.path()` method, contrary to what is stated in the meson documentation.

Reply:
Hmm, until https://github.com/mesonbuild/meson/issues/4070 is resolved, you can use less portable `python3 = find_program('python3')`. It is less portable than the meson module but more portable than `/usr/bin/env`.

Reply:
Alright. The result gets less and less readable, though.